cmd/earthly/satellite_cmds.go

```diff
@@ -387,3 +398,124 @@ func (app *earthlyApp) actionSatelliteUnselect(cliCtx *cli.Context) error {
 
 	return nil
 }
+
+func (app *earthlyApp) actionSatelliteWake(cliCtx *cli.Context) error {
+	app.commandName = "satelliteWake"
+
+	if cliCtx.NArg() != 1 {
+		return errors.New("satellite name is required")
+	}
+
+	app.satelliteName = cliCtx.Args().Get(0)
+
+	cloudClient, err := cloud.NewClient(app.cloudHTTPAddr, app.cloudGRPCAddr, app.sshAuthSock, app.authToken, app.console.Warnf)
+	if err != nil {
+		return errors.Wrap(err, "failed to create cloud client")
+	}
+
+	orgID, err := app.getSatelliteOrgID(cliCtx.Context, cloudClient)
+	if err != nil {
+		return err
+	}
+
+	out := make(chan string)
+	var wakeErr error
+	go func() { wakeErr = cloudClient.WakeSatellite(cliCtx.Context, app.satelliteName, orgID, out) }()
```

Comment:
Given the async nature of this operation, I would maybe make the function `WakeSatellite` do the goroutine spawning within it, and instead have it return the chan (instead of accepting it as a param).

It makes the code using the `WakeSatellite` function simpler.

It also makes the lifecycle of the channel self-contained in a single location (creation, feeding in data, and closing).

Reply:
Yeah that seems like a nice way to do it. I'll update the `ReserveSatellite` method to do that too

Reply:
I've updated this